src/s4u/s4u_conditionVariable.cpp

```diff
@@ -0,0 +1,36 @@
+#include "simgrid/s4u/conditionVariable.hpp"
+#include "simgrid/simix.h"
+
+using namespace simgrid;
+
+s4u::CondVariable::CondVariable()  : cond_(simcall_cond_init()){
+    
+}
+
+s4u::CondVariable::~CondVariable() {
```

Comment:
You need to [manage the references](https://github.com/adfaure/simgrid/blob/68111906f834c77ebf37048344d4970dcc44c7cf/include/simgrid/s4u/mutex.hpp#L31) when colpying/moving the object. Otherwise, the refcount will be broken in presence of copies (and result in dangling pointers).